cypress.json

```diff
@@ -0,0 +1,3 @@
+{
+  "baseUrl": "http://localhost:8080"
+}
```

Comment:
Why isn't this a part of `cypress.env.example.json`?
And can we do cypress testing only locally?
We can't do E2E testing on staging or prod?

Reply:
This is something that can't be overridden with cypress env as mentioned here.
https://docs.cypress.io/guides/guides/environment-variables#Setting

In case we need to run on a different base URL, we can control it via command line while running cypress
```
CYPRESS_BASE_URL=https://staging.app.com npx cypress open
```

Reply:
We'll run cypress on CI. That I've started working on.

Reply:
So, we'll be using both `cypress.json` and `cypress.env.json`? From the documentation page, it seemed like we should be using either one of those. Also, it's not clear to me how we are parameterising the baseUrl just yet for being used in CI. Should I wait for you to update the PR with the code required to perform CI for that?

Reply:
So Cypress.json will contain all common configurations that will be used in everyone's machine (including CI). It is similar to a file say `config.js`. Cypress env contains configurations specific to developer's machine and the values within can be different for everyone.

BaseURL is not supported in cypress env as per the documentation link I've shared above. They recommend to have a default base URL mentioned in cypress.json and in case need to be overridden, it can be done via the command (mentioned above also):
```
CYPRESS_BASE_URL=https://staging.app.com npx cypress open
```

Reply:
For CI, we may have to use the similar command instead of just running `npx cypress open` if the baseUrl is not the default one that we have in cypress.json